src/components/accordion/accordion.vue

```diff
@@ -0,0 +1,121 @@
+        </button>
+        <div class="ui-accordion--body"
+                role="presentation"
+                ref="body"
+                @transitionend="onTransitionEnd">
+            <slot name="body" :is-open="state.isOpen" :is-visible="isVisible"/>
+        </div>
+    </div>
+</template>
+
+<script>
+    import forceReflow from 'bianco.force-reflow';
+    import isMountedMixin from '../../mixins/is-mounted';
+    import bemMixin from '../../mixins/bem';
+
+    const TRANSITION_END_KEY_PROP = 'max-height';
+
+    export default {
+        mixins: [
+            isMountedMixin,
+            bemMixin('ui-accordion'),
+        ],
+        props: {
+            isOpen: {
+                type: Boolean,
+                default: false,
+            },
+            isPassive: {
+                type: Boolean,
+                default: false,
+            },
+        },
+        data() {
+            return {
+                state: {
+                    isAnimating: false,
+                    isOpen: this.isOpen,
+                },
+            };
+        },
+        computed: {
+            isVisible() {
+                return this.isClosing || this.hasFinishedOpeningAnimation;
+            },
+            hasFinishedOpeningAnimation() {
+                return this.state.isOpen && !this.state.isAnimating;
+            },
+            isClosing() {
+                return !this.state.isOpen && this.state.isAnimating;
+            },
+            rootClasses() {
+                return [
+                    this.bemFacets,
+                    this.bemIf(this.state.isAnimating, 'is-animating'),
+                    this.bemIf(this.isMounted && this.state.isOpen, 'is-open'),
+                ];
+            },
+        },
+
+        methods: {
+            change(val) {
+                this.state.isOpen = val;
+                this.state.isAnimating = true;
+                this.$nextTick(this.startAnimation);
+            },
+            startAnimation() {
+                const { body } = this.$refs;
+
+                body.style.maxHeight = `${ body.scrollHeight }px`;
+
+                // NOTE: Make the accordion animations smooth on any browser
+                if (!this.state.isOpen) {
+                    forceReflow(body);
+                    body.style.maxHeight = '0px';
+                }
+
+                this.$emit('change', this.state.isOpen);
+            },
+            onRequestChange(isOpen) {
+                this.$emit('request-change', isOpen);
+            },
+            onTransitionEnd(event) {
+                const isTargetElement = event.target === this.$refs.body;
+                const isTargetProperty = event.propertyName === TRANSITION_END_KEY_PROP;
+
+                if (!isTargetElement || !isTargetProperty) {
+                    return;
+                }
+
+                this.finishAnimation();
+                this.$emit('changed', this.state.isOpen);
+            },
+            finishAnimation() {
```

Comment:
Can you move this function up to `startAnimation`? Then we have methods and listeners packed together.

Reply:
I have removed the computed and replaced them with variables